rust/src/go/nmstate/nmstate.go

```diff
@@ -0,0 +1,199 @@
+// +build cgo
+package nmstate
+
+// #cgo CFLAGS: -g -Wall
+// #cgo LDFLAGS: -lnmstate
+// #include <nmstate.h>
+// #include <stdlib.h>
+import "C"
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+type Nmstate struct {
+	Timeout	    uint
+	LogsWritter io.Writer
+	Flags	    byte
+	State	    string
+	Checkpoint  string
```

Comment:
These two has to be arguments of their functions.

Reply:
Done!